test/devicecontroller/DefaultDeviceController.test.ts

```diff
@@ -2398,28 +2398,102 @@ describe('DefaultDeviceController', () => {
+    it('reuses the original video stream to create a transform device stream', async () => {
       const spy1 = sinon.spy(DefaultVideoTile, 'disconnectVideoStreamFromVideoElement');
       const spy2 = sinon.spy(DefaultVideoTile, 'connectVideoStreamToVideoElement');
       const gUM = sinon.spy(navigator.mediaDevices, 'getUserMedia');
+      domMockBehavior.createElementCaptureStream = new MediaStream();
+      domMockBehavior.mediaStreamTrackSettings.deviceId = stringDeviceId;
+      domMockBehavior.createElementCaptureStream.addTrack(new MediaStreamTrack());
+      await deviceController.chooseVideoInputDevice(stringDeviceId);
+      deviceController.startVideoPreviewForVideoInput(element);
+      const processor = new NoOpVideoFrameProcessor();
+      const device = new DefaultVideoTransformDevice(logger, stringDeviceId, [processor]);
+      await deviceController.chooseVideoInputDevice(device);
+      deviceController.startVideoPreviewForVideoInput(element);
+
+      expect(gUM.returnValues).to.have.lengthOf(1);
+      expect(spy1.notCalled).to.be.true;
+      expect(spy2.calledTwice).to.be.true;
+      expect(gUM.calledOnce).to.be.true;
+      spy1.restore();
+      spy2.restore();
+      gUM.restore();
+    });
+
+    it('disconnects and releases the video stream if the video element stream is different from the active video stream', async () => {
+      await deviceController.chooseVideoInputDevice(stringDeviceId);
+      deviceController.startVideoPreviewForVideoInput(element);
+      const newStringDeviceId = 'device-id-with-transformation';
+      const spy1 = sinon.spy(deviceController, 'releaseMediaStream');
+      const spy2 = sinon.spy(DefaultVideoTile, 'disconnectVideoStreamFromVideoElement');
+      const gUM = sinon.spy(navigator.mediaDevices, 'getUserMedia');
+      await deviceController.chooseVideoInputDevice(newStringDeviceId);
       deviceController.startVideoPreviewForVideoInput(element);
+
+      expect(spy1.calledOnce).to.be.true;
+      expect(spy2.calledOnce).to.be.true;
+      expect(gUM.calledOnce).to.be.true;
+      spy1.restore();
+      spy2.restore();
+      gUM.restore();
+    });
+
+    it('does not disconnect and release the video stream if the video element stream is same as the active video stream', async () => {
+      await deviceController.chooseVideoInputDevice(stringDeviceId);
+      deviceController.startVideoPreviewForVideoInput(element);
+      const spy1 = sinon.spy(deviceController, 'releaseMediaStream');
+      const spy2 = sinon.spy(DefaultVideoTile, 'disconnectVideoStreamFromVideoElement');
+      const gUM = sinon.spy(navigator.mediaDevices, 'getUserMedia');
+      await deviceController.chooseVideoInputDevice(stringDeviceId);
+      deviceController.startVideoPreviewForVideoInput(element);
+
       expect(spy1.called).to.be.false;
       expect(spy2.called).to.be.false;
-      expect(gUM.calledOnce).to.be.false;
```

Comment:
What is the assertion here for gUM? Should only be called once right? Don't think we need to call `chooseVideoInputDevice` a second time.

Reply:
I agree that gUM assertion is not great. I have changed the assertion from `expect(gUM.calledOnce).to.be.false;` to `expect(gUM.called).to.be.false;`. Since, we do not call `chooseVideoInputDevice()` gUM should not be called.